mantidproject/mantid
# Support IN16B's single detectors in IndirectILLEnergyTransfer

**Description of work.**
In BATS mode, IN16B can have single detectors, that needs the same treatments as the other detectors, but without grouping.
This PR takes care of this, and fixes an indexation bug for regular pixel grouping.



**To test:**
- Run
 ```
center, epp = IndirectILLEnergyTransfer(Run='215962', PulseChopper='34', GroupDetectors=False)

offset_p100 = IndirectILLEnergyTransfer(Run='215967', InputElasticChannelWorkspace=epp, PulseChopper='34', GroupDetectors=False)
```
(with data from `ExternalData/Testing/SystemTest/ILL/IN16B`)
- Plot the color fill of the results and check there is an elastic peak neatly aligned.
- Plot spectra 2050 and 2051, and check the elastic peak is correctly centered on 0. 

WIP for #30591 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
